src/plot_hourly_line_graph.py

```diff
@@ -3,32 +3,61 @@
 import datetime
 import csv
 import numpy
+import os
+
+directory = r"..\graphs\hourly_line_graph"
+if not os.path.exists(directory):
+    os.makedirs(directory)
 
 list = []
-x = []
-y = []
-bins_x = []
 
 with open('../data/data.csv', 'r') as csvfile:
-    plots = csv.reader((l.replace('\0', '') for l in csvfile))
+    plots = csv.reader(csvfile, delimiter=',')
     for row in plots:
         try:
             list.append([datetime.datetime.strptime(
-                row[0], '%m/%d/%y %H:%M'), int(row[1])])
+                row[0], '%Y-%m-%d %H:%M:%S.%f'), int(row[1])])
         except:
             pass
 
-bins_x = []
-bins_y = []
+bins_x = set()
+date_dict = {}
+start_count = None
```

Comment:
I'm pretty sure its not a good practice to instantiate something to `None`. Also, it may be more descriptive to say its the start value for a particular date. 

`start_date` is used to keep track of the current date so change it to `current_date`

```
date_start_count = 0
current_date = datetime.time()
```